src/decorators.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { StoryFn, StoryContext } from '@storybook/addons';
+import { PARAM_KEY } from './constants';
+
+export const withApolloClient = (
+  storyFn: StoryFn,
+  context: StoryContext
+): JSX.Element => {
+  const { MockedProvider, ...providerProps } =
+    context.parameters[PARAM_KEY] ?? {};
+
+  if (!MockedProvider) {
+    console.warn(
+      'storybook-addon-apollo-client: MockedProvider is missing from parameters in preview.js'
+    );
+
+    return storyFn() as JSX.Element;
+  }
+
+  return (
+    <MockedProvider {...providerProps}>
+      {storyFn(context) as JSX.Element}
```

Comment:
Nit: in v6 Storybook moved to using `<Story />` component syntax in decorators
https://storybook.js.org/docs/react/writing-stories/decorators#global-decorators

In my experience, using the `storyFn()` format will cause the story code to be executed once before the parent components/context providers are initialized, which can introduce bugs that are difficult to track down